src/url.ts

```diff
@@ -1,23 +1,27 @@
 import * as t from "io-ts";
 import * as url from "url";
 
-const isUrl = (v: t.mixed): v is url.Url =>
-  v !== null && typeof v === "object" && "href" in v;
+export type ValidUrl = url.Url & { readonly href: string };
```

Comment:
isn't `Url` supposed to describe a valid URL already? also `Url` already defines an [href attribute](https://nodejs.org/api/url.html#url_url_href)

Reply:
nope. Url defines an optional href attribute which maybe undefined under some circumstances). 
we generally deal with http(s) url, so we need href to be always defined (that's the reason of this check). UrlFromString is a subset of all possible Url(s)